opencti-platform/opencti-front/src/components/EditEntityButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Button, styled } from '@mui/material';
+import { Edit } from '@mui/icons-material';
+import { useFormatter } from './i18n';
+
+interface EditEntityControlledDialOptions {
+  size?: 'small' | 'medium' | 'large'
+  variant?: 'text' | 'contained' | 'outlined'
+  margin?: string
+}
+
+/**
+ * Get a generator for an entity edit button
+ * @param opts.size size of the button and its icon
+ * @param opts.variant type of button
+ * @param opts.margin css margin. defaults to 3px on left
+ * @returns Generator that takes an onOpen function and returns a button
+ */
+function EditEntityControlledDial({
```

Comment:
I do not understand why you are using a generator function.
React components are (or should be) generic by design.

If it's only because of typescript errors, I'm sure we can find a solution, maybe make the component type-generic.